### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by R-Munitz**

### CODE REVIEW

**Comment by andrewoid** [comment]
Add this to your list of dependencies in build.gradle
`    testCompile "org.testfx:testfx-junit5:4.0.16-alpha"`

Add this to your MinesweeperControllerTest
``` java
@ExtendWith(ApplicationExtension.class)
public class MinesweeperControllerTest {
```

**Comment by R-Munitz** [comment]
That worked, thanks!

File: src/main/java/MinesweeperController.java [hunk 1]
```diff
@@ -116,20 +116,20 @@ private boolean isOnGrid (int x, int y){
...
         ArrayList<MinesweeperCell> adjacentCells = new ArrayList<>();
-        //fix labels, y - axis in descending order
-        // top left
+        // bottom left
         if (isOnGrid(x-1,y+1)){
             adjacentCells.add(board[x - 1][y + 1]);
         }
-        //top middle
+        //bottom middle
         if (isOnGrid(x,y+1)){
```
Comment (andrewoid) [inline src/main/java/MinesweeperController.java]: There is a lot of repetitive code here. 
Can you shorten it?

File: src/test/java/MinesweeperControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            public void run() {
+                Application.launch(FXRunnerApplication.class);
+            }
+        }.start();
+    }
+    public static class FXRunnerApplication extends Application {
+        @Override
+        public void start(Stage paramStage) throws Exception {
+        }
+    }
```
Comment (andrewoid) [inline src/test/java/MinesweeperControllerTest.java]: With the `@ExtendWith` this shouldn't be necessary anymore

File: src/test/java/MinesweeperControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        expectedAdjacentCells.add(new MinesweeperCell(1,1)); //bottom middle;
+        expectedAdjacentCells.add(new MinesweeperCell(2,0)); //right
+        expectedAdjacentCells.add(new MinesweeperCell(2,1)); //bottom right
+
+
+        //when
+        ArrayList<MinesweeperCell> adjacentCells = controller.getAdjacentCells(sampleCell);
+
+        //then
+        //is this using the MinesweeperCell equals method?
```
Comment (andrewoid) [inline src/test/java/MinesweeperControllerTest.java]: This is using the equals() method however your equals method checks for more than just the x,y coordinates.

File: src/test/java/MinesweeperControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        assertEquals(adjacentCells,expectedAdjacentCells);
+
+    }
+
+    private void givenMinesweeperController() {
+        controller = new MinesweeperController();
+        // should this code go here?
+        for (int x = 0; x < boardSize; x++){
+            for(int y = 0; y < boardSize; y++){
+                board[x][y] = mock(MinesweeperCell.class);
```
Comment (andrewoid) [inline src/test/java/MinesweeperControllerTest.java]: You're instantiating a board that is never actually used.

File: src/test/java/MinesweeperControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        //verify addOnClick method called, and works
+
+    }
+
+    @Test
+    public void getAdjacentCells(){ //passes sometimes?
+        //given
+        givenMinesweeperController();
+        ArrayList<MinesweeperCell> expectedAdjacentCells = new ArrayList<>();
+        MinesweeperCell sampleCell = new MinesweeperCell(1,0);
```
Comment (andrewoid) [inline src/test/java/MinesweeperControllerTest.java]: Use the actual cells from the Controller's board.

Comment (R-Munitz) [reply src/test/java/MinesweeperControllerTest.java]: I created those as mocks though, should I not have?

Comment (andrewoid) [reply src/test/java/MinesweeperControllerTest.java]: but the Controller is not using the mocks.

Comment (R-Munitz) [reply src/test/java/MinesweeperControllerTest.java]: thought I would need them for other methods. changed it in the latest commit.

File: src/test/java/MinesweeperControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        controller.initialize();
+
+        //then
+        //verify cells placed on gridPane
+        //verify addOnClick method called, and works
+
+    }
+
+    @Test
+    public void getAdjacentCells(){ //passes sometimes?
```
Comment (andrewoid) [inline src/test/java/MinesweeperControllerTest.java]: Does this still only pass sometimes?

Comment (R-Munitz) [reply src/test/java/MinesweeperControllerTest.java]: no, forgot to remove the comment.

**Pull request merged**